### DISCUSSION TITLE
Reasoning for static library

### DISCUSSION THREAD

**Discussion by kayaercument** [Munich-Quantum-Software-Stack/QDMI#48] [Q&A] [answered]
Hello @burgholzer 👋 

Thank you for your work on the repository! I realize that the QDMI library is a static library instead of a shared. Is there any particular reason for this?

**Comment by burgholzer** [comment, marked as answer]
Hey 👋🏼 

It is not fixed to be a static library. You can simply pass `-DBUILD_SHARED_LIBS=ON` to the CMake configure step and it will happily build a shared library.
See https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html for more information.
This allows for quite some additional flexibility in the build system.
If it turns out we do not need that flexibility and we always want to build a shared library anyway, we can always add that back in.

**Reply by kayaercument** [reply]
Thank you for the explanation!

We had some problems while handling QDMI as a dependency. The solution is putting `set(BUILD_SHARED_LIBS ON)`  after `FetchContent_Declare `.

**Reply by burgholzer** [reply]
That certainly should work.
However, if depending projects do not build with QDMI being built as a static library, there is no added value by the flexibility and I would rather just mark the QDMI library as shared again.
I'll submit a PR later on 👍🏻
